packages/firestore/src/core/firestore_client.ts

```diff
@@ -536,7 +536,10 @@ export class FirestoreClient {
   }
 
   unlisten(listener: QueryListener): void {
-    this.verifyNotShutdown();
+    // Make unlisten a no-op if client is shut down.
```

Comment:
Your code is so clear that I think we can remove this comment :)

Reply:
I actually want to keep it. The reason is `FirestoreClient` methods generally run `verifyNotShutdown` before doing anything. This one deviate from that. 

I updated the comment to be more clear.